app/graphql/mutations/archive_course.rb

```diff
@@ -0,0 +1,54 @@
+module Mutations
+  class ArchiveCourse < ApplicationQuery
+    include QueryAuthorizeSchoolAdmin
+    class CourseMustNotBeArchived < GraphQL::Schema::Validator
+      def validate(_object, _context, value)
+        course = Course.find_by(id: value[:id])
+
+        return "Unable to find course with id: #{value[:id]}" if course.blank?
+
+        return 'Course is already archived' if course.archived?
+      end
+    end
+
+    argument :id, ID, required: true
+
+    description 'Archives a course.'
+
+    validates CourseMustNotBeArchived => {}
+
+    field :success, Boolean, null: false
+
+    def resolve(_params)
+      archive_course
+      notify(
+        :success,
+        I18n.t('shared.done_exclamation'),
+        I18n.t('mutations.archive_course.success_notification')
+      )
+      { success: true }
+    end
+
+    def archive_course
```

Comment:
Would be nice to add domain events for each new feature (and publish them using dev's notification service)

Reply:
Done @mpraglowski